src/CommunityToolkit.Maui.BindablePropertySG/BindablePropertySG.cs

```diff
@@ -0,0 +1,192 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CommunityToolkit.Maui.BindablePropertySG;
+
+[Generator]
+public class BindablePropertySG : IIncrementalGenerator
+{
+	static readonly SemanticValues emptySemanticValues = new(default, Array.Empty<BPInfo>().ToImmutableArray());
+
+	const string bpFullName = "global::Microsoft.Maui.Controls.BindableProperty";
+	const string bindingModeFullName = "global::Microsoft.Maui.Controls.";
+
+	const string bpAttribute = """
+#nullable enable
+namespace CommunityToolkit.Maui.BindablePropertySG;
+
+[AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
+public sealed class BindablePropertyAttribute<TReturnType> : Attribute
+{
+	public string PropertyName { get; } = string.Empty;
+	public Type? OwnerType { get; set; }
+	public object? DefaultValue { get; set; }
+	public string DefaultBindingMode { get; set; } = string.Empty;
+	public string ValidateValueMethodName { get; set; } = string.Empty;
+	public string PropertyChangedMethodName { get; set; } = string.Empty;
+	public string PropertyChangingMethodName { get; set; } = string.Empty;
+	public string CoerceValueMethodName { get; set; } = string.Empty;
+	public string DefaultValueCreatorMethodName { get; set; } = string.Empty;
+
+	public BindablePropertyAttribute(string propertyName)
+	{
+		PropertyName = propertyName;
+	}
+}
+""";
+
+	public void Initialize(IncrementalGeneratorInitializationContext context)
+	{
+		context.RegisterPostInitializationOutput(ctx => ctx.AddSource("BindablePropertyAttribute.g.cs", SourceText.From(bpAttribute, Encoding.UTF8)));
+
+		var provider = context.SyntaxProvider.ForAttributeWithMetadataName("CommunityToolkit.Maui.BindablePropertySG.BindablePropertyAttribute`1",
+			SyntaxPredicate, SemanticTransform)
+			.Where(static x => x.ClassInformation != default || !x.BPInfos.IsEmpty)
+			.Collect()
+			.SelectMany(static (types, _) => types);
+
+
+		context.RegisterSourceOutput(provider, Execute);
+	}
+
+	void Execute(SourceProductionContext context, SemanticValues semanticValues)
+	{
+		var source = GenerateSource(semanticValues);
+		SourceStringService.FormatText(ref source);
+		context.AddSource($"{semanticValues.ClassInformation.ClassName}.g.cs", SourceText.From(source, Encoding.UTF8));
+	}
+
+	static string GenerateSource(SemanticValues value)
+	{
+		var sb = new StringBuilder($@"
+// <auto-generated>
+// Test2 : {DateTime.Now}
+
+namespace {value.ClassInformation.ContainingNamespace};
+
+{value.ClassInformation.DeclaredAccessibility} partial class {value.ClassInformation.ClassName}
+{{
+");
+
+		foreach (var info in value.BPInfos)
+		{
+			GenerateBindableProperty(sb, info);
+			GenerateProperty(sb, info);
+		}
+
+		sb.AppendLine().Append("}");
+		return sb.ToString();
+
+		static void GenerateBindableProperty(StringBuilder sb, BPInfo info)
+		{
+			/// <summary>
+			/// Backing BindableProperty for the <see cref="PropertyName"/> property.
+			/// </summary>
```

Comment:
For us this will work very well, not sure how to improve this when it goes public.